### PR TITLE
R GPU Dockerfiles

### PR DESCRIPTION
**Pull request by brnleehng**
- Added shared library to base Dockerfile
- Added Dockerfiles:
-- Change Pyenv to writable folder
-- Preinstall keras and tensorflow R packages
-- Added openmpi-bin (Required for CNTK)
-- Added path environment variable into R environment ($PATH is not recognized in Rstudio Server)

### CODE REVIEW

File: docker-image/r/spark2.2.0/gpu/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+RUN  useradd -m -d /home/rstudio rstudio -G sudo,staff \
+  && echo rstudio:rstudio | chpasswd \
+  && chmod -R 777 /home/rstudio \
+  && chmod -R 777 //.pyenv/
+
+# Setting up rstudio user with Tensorflow and CNTK
+USER rstudio
+RUN  echo "PATH='"$PATH"'" > /home/rstudio/.Renviron \
+  && pip3 install \
+    $CNTK_GPU_VERSION \
```
Comment (paselem) [inline docker-image/r/spark2.2.0/gpu/Dockerfile]: any reason we're not using the same variable definition for tensorflow and keras?

File: docker-image/r/spark2.2.0/gpu/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    make \
+    unzip \
+    zip \
+    libcurl4-openssl-dev \
+    libxml2-dev \
+    libapparmor1 \
+    gdebi-core \
+    lsb-release \
+    psmisc \
+    sudo \
```
Comment (paselem) [inline docker-image/r/spark2.2.0/gpu/Dockerfile]: is sudo required?

File: docker-image/r/spark2.2.0/gpu/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+               --enable-memory-profiling \
+               --with-readline \
+               --with-blas="-lopenblas" \
+               --disable-nls \
+               --without-recommended-packages \
+  ## Build and install
+  && make \
+  && make install \
+  ## Add a default CRAN mirror
+  && echo "options(repos = c(CRAN = 'https://cran.rstudio.com/'), download.file.method = 'libcurl')" >> /usr/local/lib/R/etc/Rprofile.site \
```
Comment (paselem) [inline docker-image/r/spark2.2.0/gpu/Dockerfile]: super nitpick: line too long (there are a few of these)

File: docker-image/r/spark2.2.0/gpu/Dockerfile (hunk 2 above)
Comment (brnleehng) [reply docker-image/r/spark2.2.0/gpu/Dockerfile]: sudo required for rstudio server

**Pull request merged**